src/main/kotlin/snyk/iac/annotator/IacBaseAnnotator.kt

```diff
@@ -0,0 +1,66 @@
+package snyk.iac.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.ExternalAnnotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.psi.PsiFile
+import io.snyk.plugin.ui.toolwindow.SnykToolWindowPanel
+import snyk.iac.IacIssue
+import snyk.iac.IacResult
+
+private val LOG = logger<IacBaseAnnotator>()
+
+abstract class IacBaseAnnotator : ExternalAnnotator<PsiFile, List<IacIssue>>() {
```

Comment:
ordinary Annotator fit better as we don't do external calls

Reply:
if we keep using externalAnnotator then for readability methods order need to mutch the execution (parent class) order

Reply:
External Annotator looks good in this case - let's stick to it for now. The annotations are provided from an external tool after-all, even if we cache it. Having jetbrains handle the complexity is a benefit on top :).

I will rearrange the order though, to match your preference :)

Reply:
External Annotator should be used if actions from outside world is needed. In our case we took all what we need form cache only, so absolutely no needs in External Annotator in our case.